File: backend/domain/question.py
from dataclasses import dataclass, field
from enum import Enum
from uuid import UUID, uuid4


class QuestionType(Enum):
    HR = "HR"
    PRIMARY = "PRIMARY"
    FOLLOW_UP = "FOLLOW_UP"


@dataclass
class Question:
    id: UUID = field(default_factory=uuid4)
    text: str = ""
    question_type: QuestionType = QuestionType.PRIMARY
    order: int = 0

File: backend/src/models/question.js
import { randomUUID } from "node:crypto";
import { QuestionType } from "./question-type.js";

export class Question {
  constructor(opts) {
    this.id = opts?.id ?? randomUUID();
    this.text = opts?.text ?? "";
    this.questionType = opts?.questionType ?? QuestionType.PRIMARY;
    this.order = opts?.order ?? 0;
  }
}
